llvm/include/llvm/Target/Target.td

```diff
@@ -583,11 +583,15 @@ class EncodingByHwMode<list<HwMode> Ms = [], list<InstructionEncoding> Ts = []>
   list<InstructionEncoding> Objects = Ts;
 }
 
+class RegClass<RegisterClass Reg, int OperIdx = 0> {
```

Comment:
Perhaps `class OperandRegClass` and reverse the parameters?

Reply:
I agree, always having an operand idx is clearer when reading the .td